### ISSUE TITLE
The Lora effect is not noticeable

### ISSUE THREAD

**Issue by zhang17628** [ace-step/ACE-Step-1.5#728]
It's really good work, recently I've been fine-tuning ACE-step, fine-tuning 500 epochs with 400 songs or so. But I found that the style change before and after fine-tuning was not noticeable. Can you give me some advice?
Here is the song before lora:

**Comment by 1larity** [comment]
Ensure that "thinking" and other COT settings are turned off .
Make sure training and generation models are the same, if you trained on turbo, use turbo model for generation.

These really should be set automatically for LoRA generation, but compared to other stuff on the todo list, it's minor.

**Comment by Huli-fox** [comment]
> Ensure that "thinking" and other COT settings are turned off . Make sure training and generation models are the same, if you trained on turbo, use turbo model for generation.
> 
> These really should be set automatically for LoRA generation, but compared to other stuff on the todo list, it's minor.

I have a different opinion on the second point. I trained on base and it works well on turbo.

**Comment by Helikaon23s** [comment]
> Ensure that "thinking" and other COT settings are turned off . Make sure training and generation models are the same, if you trained on turbo, use turbo model for generation.
> 
> These really should be set automatically for LoRA generation, but compared to other stuff on the todo list, it's minor.

Do you mean to turn those settings off for making the lora or for making music ? sorry, I'm confused

**Comment by Helikaon23s** [comment]
> > Ensure that "thinking" and other COT settings are turned off . Make sure training and generation models are the same, if you trained on turbo, use turbo model for generation.
> > These really should be set automatically for LoRA generation, but compared to other stuff on the todo list, it's minor.
> 
> I have a different opinion on the second point. I trained on base and it works well on turbo.

From the chats on Ace-Step's Discord , the understanding appears to be that Base is the one rin..... model to rule them all , if it's made in Base, then you can use it in SFT , Turbo and of course Base , but it doesn't work the other way.

**Comment by lbryant2004-create** [comment]
> > Ensure that "thinking" and other COT settings are turned off . Make sure training and generation models are the same, if you trained on turbo, use turbo model for generation.
> > These really should be set automatically for LoRA generation, but compared to other stuff on the todo list, it's minor.
> 
> Do you mean to turn those settings off for making the lora or for making music ? sorry, I'm confused

Hi, it works for me after I load the LoRA and disable thinking and other COT settings when I want to generate music using the LoRA. Mine still needs training, as it is quite noisy with the SFT model after 200 epochs of training and 107 songs of varying styles, but I think we'll get there after epoch 500 or so. Does anyone get "nonfinite gradients" during training? It doesn't seem to affect learning as the loss does eventually decrease.

**Comment by Helikaon23s** [comment]
> > > Ensure that "thinking" and other COT settings are turned off . Make sure training and generation models are the same, if you trained on turbo, use turbo model for generation.
> > > These really should be set automatically for LoRA generation, but compared to other stuff on the todo list, it's minor.
> > 
> > 
> > Do you mean to turn those settings off for making the lora or for making music ? sorry, I'm confused
> 
> Hi, it works for me after I load the LoRA and disable thinking and other COT settings when I want to generate music using the LoRA. Mine still needs training, as it is quite noisy with the SFT model after 200 epochs of training and 107 songs of varying styles, but I think we'll get there after epoch 500 or so. Does anyone get "nonfinite gradients" during training? It doesn't seem to affect learning as the loss does eventually decrease.

That's strange, I turn them all off and the audio just turns to mush . Good output with them on, poor off.

**Comment by lbryant2004-create** [comment]
> > > > Ensure that "thinking" and other COT settings are turned off . Make sure training and generation models are the same, if you trained on turbo, use turbo model for generation.
> > > > These really should be set automatically for LoRA generation, but compared to other stuff on the todo list, it's minor.
> > > 
> > > 
> > > Do you mean to turn those settings off for making the lora or for making music ? sorry, I'm confused
> > 
> > 
> > Hi, it works for me after I load the LoRA and disable thinking and other COT settings when I want to generate music using the LoRA. Mine still needs training, as it is quite noisy with the SFT model after 200 epochs of training and 107 songs of varying styles, but I think we'll get there after epoch 500 or so. Does anyone get "nonfinite gradients" during training? It doesn't seem to affect learning as the loss does eventually decrease.
> 
> That's strange, I turn them all off and the audio just turns to mush . Good output with them on, poor off.

That's happening to me as well, however I have been testing each time I get a new checkpoint by running another copy of ACE-Step on port 7861 and the results seem to get a tiny bit better, which suggests my LoRA still needs to train. It's been training now for about five days, LOL. Could be the model though so i might try with others, like base.

**Comment by Kwisss** [comment]
as far as I know the original (gradio) trainer would never work on the base or sft model,
the v2 should do something, but with 400 plain songs with lyrics you really need to a/b the result with a locked seed without the llm.

I tried training the sft model, with the muon optimizer on some layers with the cfg dropout and timestep code from v2, it responds decently well but my current database toasts the instrumentation in the background.

Right now I am in the process of making a copy and converting 10% of the current database in instrumentals with romofor, and adding captions that reflects that. I will also 10% unique instrumentals with captions.

about the "tests" others are running on the sft and the base model, I actually only noticed a loss graph slowly coming down when it is over-training?
Would be happy to hear back from you guys how it worked out for you? The longest run I found useful so far for the base and sft model is about 80 epochs? after that I would categorize it as unusable on 0.8 strength even?

I also had less success with the higher ranks (128/256) and doubling the alpha also does not seem work out for the base/sft model

**Comment by 1larity** [comment]
Apocryphally, I've had more luck with SFT and Turbo (a little noisy but acceptable for quick gens). With base it seems to ignore my vocal style and I get the American guy, and the music is a bit garbled*. Side step is way faster for LoRA training, ~3s per epoch, compared to 90s per epoch with the native ACE-step LoRA training. 3060ti

Tested 1k epochs on SFT, Base, turbo with both sidestep and ace step training. 8 songs, only 2 with instruments, 500-700 seems to be my sweet spot with .85 scale.

*Chat GPT informs me that bases models naturally need a lot more samples and training than SFT/turbo, and 8 songs with 1000 epoch may not be enough.

**Comment by shawpook** [comment]
So the conclusion is that no one seems to have achieved good results in Lora training, and I'm no exception.

**Comment by Helikaon23s** [comment]
> So the conclusion is that no one seems to have achieved good results in Lora training, and I'm no exception.

I've made about 15 so far and produced good results . This is how I make them : https://github.com/Grey3016/ACE-Step-TrainingSetupV1.0

A more detailed guide is here :  https://github.com/sdbds/ACE-Step-1.5-for-windows/blob/main/docs/en/LoRA_Training_Tutorial.md

**Comment by AtlasRedux** [comment]
> So the conclusion is that no one seems to have achieved good results in Lora training, and I'm no exception.

I have made several LoRAs, all that had fantastic results.
The issues are, as always, in front of the screen.

> > Ensure that "thinking" and other COT settings are turned off . Make sure training and generation models are the same, if you trained on turbo, use turbo model for generation.
> > These really should be set automatically for LoRA generation, but compared to other stuff on the todo list, it's minor.
> 
> I have a different opinion on the second point. I trained on base and it works well on turbo.

Same. And merged.

**Comment by shawpook** [comment]
> > 因此，结论是似乎没有人能在 Lora 训练中取得好成绩，我也不例外。
> 
> 我已经制作了大约 15 个，效果不错。这是我的制作方法：https://github.com/Grey3016/ACE-Step-TrainingSetupV1.0
> 
> 更详细的指南请参见： https://github.com/sdbds/ACE-Step-1.5-for-windows/blob/main/docs/en/LoRA_Training_Tutorial.md

Thank you, I will study right away.

**Comment by shawpook** [comment]
> > 因此，结论是似乎没有人能在 Lora 训练中取得好成绩，我也不例外。
> 
> 我已经制作了大约 15 个，效果不错。这是我的制作方法：https://github.com/Grey3016/ACE-Step-TrainingSetupV1.0
> 
> 更详细的指南请参见： https://github.com/sdbds/ACE-Step-1.5-for-windows/blob/main/docs/en/LoRA_Training_Tutorial.md

My model is effective, but I'm not particularly satisfied with it. I've adjusted it in many ways, but I still don't know what the optimal approach is. Regarding what you mentioned, "it auto stops when it sees that the epochs aren't getting better," I often see it stop at 200 epochs. I thought this was a bug; could it be that this is normal? 200 is too few.

**Comment by Helikaon23s** [comment]
> > > 因此，结论是似乎没有人能在 Lora 训练中取得好成绩，我也不例外。
> > 
> > 
> > 我已经制作了大约 15 个，效果不错。这是我的制作方法：https://github.com/Grey3016/ACE-Step-TrainingSetupV1.0
> > 更详细的指南请参见： https://github.com/sdbds/ACE-Step-1.5-for-windows/blob/main/docs/en/LoRA_Training_Tutorial.md
> 
> My model is effective, but I'm not particularly satisfied with it. I've adjusted it in many ways, but I still don't know what the optimal approach is. Regarding what you mentioned, "it auto stops when it sees that the epochs aren't getting better," I often see it stop at 200 epochs. I thought this was a bug; could it be that this is normal? 200 is too few 

Different styles of music , datasets and settings will change the epochs required, the trainer I mentioned has a setting on it to stop after the default setting of 80 epochs without improvement. This can be right or wrong depending on the music and datasets . Yes, 200 seems too few but it can depend on the number of tracks in your dataset and styles, the trainer is trying to find similarities, if the music is very similar then it would arguably finish with far fewer epochs,

My general approach with my understanding is 
1.Use flacs or wavs to make your dataset - the trainer can/will convert them to 48kHz wavs & they should be normalised as whole folder
2.Go through the tracks and delete any that are outliers (eg a country song in a folder of house music)
3.Personally I go for 9-11 tracks and my loras have turned out ok - this trains to the style of the album
4.The (description) caption  : ANY caption is better than no caption (and I have quite a few loras with shit captions that make good tracks) , which also implies that the better the caption, the better the outcomes
5.How do you get better descriptions ? I'll shorthand this bit to save my sanity, lyrics: https://genius.com/ , key & bpm: https://songbpm.com/ , caption: https://huggingface.co/spaces/nvidia/music-flamingo
6.When making the lora - I make mine like this : https://github.com/Grey3016/ACE-Step-TrainingSetupV1.0 , other trainers are available , as are other settings 
7.When making the song - turn on 'Think' , it helps with song structure and helps stop musical discordancy 
8.When making the song - overlong length leads to quality drop, I keep mine to about 3 1/2mins
9.When making the song - some lyrics (or how they're written) can break the song by just not fitting, sometimes this is a seed issue 
10.Finally , when making the song - use a good detailed prompt, if you know the sound you want and you've made a good lora dataset json (ie good captions) , borrow the descriptions from there and make a hybrid prompt
11.Keep your dataset jsons for the above reason. 

I've changed my way of training over the last day, inspecting my manifest file it appears that the trainer was not adding in the caption to the file and making all the files instrumental , so I'm (as I type) using a new trainer , Side-Step . It's on epoch 208/1200 , so I have a couple of hours until I can give feedback on this.

**Comment by shawpook** [comment]
@Grey3016 Yes, I also tried Side-Step. Currently, when using it for a certain singer's songs with 50 songs for learning, the result is quite good. But for a specific style, I selected about 200 popular songs of that style, and the final effect was very mediocre. I tested over a hundred generated songs, and none of them had a Special and excellent  melody.

**Comment by Meliauo** [comment]
If full fine-tuning is possible, I believe it would yield better results.

**Comment by Helikaon23s** [comment]
> [@Grey3016](https://github.com/Grey3016) Yes, I also tried Side-Step. Currently, when using it for a certain singer's songs with 50 songs for learning, the result is quite good. But for a specific style, I selected about 200 popular songs of that style, and the final effect was very mediocre. I tested over a hundred generated songs, and none of them had a Special and excellent melody.

On one had I'd suggest trying less songs for a style and within a smaller spread of variation , if it is dance music for example, while most (if not all of it) is house music at its core , if you put in trance with house with techno with dubstep , the trainer won't know what the hell is going on . 

 Side-Step has now been updated to v1 , some of the enhancements (settings) let you focus on style but you'd need to go through it all as there are a lot of settings in it.
